lib/wlib/memory/Memory.h

```diff
@@ -310,7 +316,7 @@ Type *realloc(Type *ptr, wlp::size_type num = 1) {
     wlp::size32_type oldMemSize = getFixedMemorySize(ptr);
     wlp::size32_type newMemSize = static_cast<wlp::size32_type>(sizeof(Type)) * num;
 
-    if (oldMemSize == newMemSize) {
+    if (oldMemSize >= newMemSize + getSmallestBlockSize()) {
         return ptr;
     }
 

```

Comment:
A little bit down in the code from here (git doesn't let me comment on the exact line) you do a for loop over byte  (`char`) arrays to copy the data. Would be faster to use `memcpy` because it's one of those super optimized functions.